docs: update example os listing users with python client

plaguss opened the pull request:
<!-- Thanks for your contribution! As part of our Community Growers initiative 🌱, we're donating Justdiggit bunds in your name to reforest sub-Saharan Africa. To claim your Community Growers certificate, please contact David Berenstein in our Slack community or fill in this form https://tally.so/r/n9XrxK once your PR has been merged. -->

# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Closes #3453

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (change restructuring the codebase without changing functionality)
- [ ] Improvement (change adding some improvement to an existing functionality)
- [x] Documentation update

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

- [ ] Test A
- [ ] Test B

**Checklist**

- [ ] I added relevant documentation
- [ ] follows the style guidelines of this project
- [ ] I did a self-review of my code
- [x] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

plaguss committed:
docs: update example os listing users with python client

plaguss committed:
Merge branch 'develop' of https://github.com/argilla-io/argilla into develop

nataliaElv commented on docs/_source/tutorials/notebooks/labelling-tokenclassification-basics.ipynb:
```diff
@@ -198,6 +187,36 @@
+    "\n",
+    "# get the list of users\n",
+    "# optional: filter users to get only those with annotator role\n",
+    "users = [u for u in rg.User.list() if u.role == \"annotator\"]"
```
The following suggestion addresses these issues:
- Listing users is limited to the role `owner`, but perhaps an `admin` might be doing the assignments. 
- We want to make sure that we only make assignments to users that have access to the dataset.
```suggestion
    "# get workspace where the dataset is (or will be) located",
    "ws = rg.Workspace.from_name("my_workspace")",
    "# get the list of users with access to the workspace\n",
    "# make sure that all users that will work on the dataset have access to the workspace\n",
    "# optional: filter users to get only those with annotator role\n",
    "users = [u for u in ws.users if u.role == \"annotator\"]"
```

nataliaElv commented on docs/_source/tutorials/notebooks/labelling-tokenclassification-basics.ipynb:
```diff
@@ -198,6 +187,36 @@
+    "    api_url=\"...\",\n",
+    "    api_key=\"...\"\n",
+    ")\n",
+    "\n",
```
These lines aren't necessary because the import and `rg.init` are present in the first blocks of the tutorial.

plaguss replied:
Sorry, I should have noted that, thanks for the review.

plaguss committed:
Update docs/_source/tutorials/notebooks/labelling-tokenclassification-basics.ipynb

The following suggestion addresses these issues:
- Listing users is limited to the role `owner`, but perhaps an `admin` might be doing the assignments.
- We want to make sure that we only make assignments to users that have access to the dataset.

Co-authored-by: Natalia Elvira <126158523+nataliaElv@users.noreply.github.com>

plaguss committed:
Merge branch 'develop' of github.com:plaguss/argilla into develop

plaguss committed:
docs: remove unnecessary lines and address issues related to roles and users' access

plaguss committed:
docs: fix example on accessing the username field of a User
